sw/device/lib/dif/README.md

```diff
@@ -60,17 +108,17 @@ there are some relaxations of these rules for them described at the end.
     is for an optional out-parameter. Arguments typed `mmio_region_t` are not
     pointers, and cannot meaningfully be checked for non-nullness.
   * DIFs must ensure, if they only accept a subset of an enum, that the argument
-    is within that subset. However, DIFs should assume, for checking
-    preconditions, that any enum argument is one of the enum variants.
-  * DIFs must not have side-effects before any guard statements. Side-effects
-    include (but are not limited to) writing to memory, including memory-mapped
-    hardware, and modifying processor CSRs.
+    is within that subset. However, DIFs may assume, for checking preconditions,
+    that any enum argument is one of the enum constants.
+  * DIFs must not cause any side-effects before any guard statements.
 
 * Switch statements in DIFs must always have a default case, including when
   switching on an enum value (an "enum switch").
   * The default case of an enum switch must report an error for values that are
-    not an enum variant.
-  * Enum switches do not need a `case` for enum variants that are unreachable
+    not a constant from that enum. In the absence of more specific information,
+    this must return `kDif<ip>Error` or the equivalent return code value from
+    a more specific return code enum.
```

Comment:
Nit: include a caveat that `kDif<ip>BadArg` may be returned if the switch is part of a precondition check.

Reply:
Yeah, re-reading this it shouldn't be a `must` sentence.